src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs

```diff
@@ -54,16 +55,18 @@ public bool Validate(BlockHeader header, bool isUncle)
     /// </returns>
     public bool ValidateSuggestedBlock(Block block)
     {
-        Transaction[] txs = block.Transactions;
         IReleaseSpec spec = _specProvider.GetSpec(block.Header);
 
-        for (int i = 0; i < txs.Length; i++)
+        if (!ValidateTransactions(block, spec, out int blobsInBlock, out string error))
         {
-            if (!_txValidator.IsWellFormed(txs[i], spec))
-            {
-                if (_logger.IsDebug) _logger.Debug($"{Invalid(block)} Invalid transaction {txs[i].Hash}");
-                return false;
-            }
+            if (_logger.IsDebug) _logger.Debug($"{Invalid(block)} {error}");
+            return false;
+        }
+
+        if (!ValidateDataGasUsed(block, spec, blobsInBlock, out string dataGasError))
+        {
+            if (_logger.IsDebug) _logger.Debug($"{Invalid(block)} {dataGasError}");
```

Comment:
minor: move the logging into the method and simplify the method - no out parameter?